### PR TITLE
OY2-16335: Add Package Actions for Base Waivers and Renewal Waivers: Temporary Extension

### PR DESCRIPTION
**Pull request by kristin-at-theta**
Story: https://qmacbis.atlassian.net/browse/OY2-16335
Endpoint: https://d1cs35nowomkub.cloudfront.net/

### Details
Add the "Request a Temporary Extension" action to the Action Popups and Detail page Links

### Changes
- moved ConfirmationDialog to be able to be used through the appContext (leaving it as-is in the PROD areas)
- updated Package Pages to use context confirmations
- pulled Action management out of containing pages, combined with the "popup" code and created ActionPopup
- added "Request Temporary Extension" as a package action that goes to the temporary extension page with the caller's id listed as the parent.

### Implementation Notes
Fixed a bug in the v0 versioning - the v1's and v2's were missing some of the fields.
Clean up - forms do not use actionsByStatus, so removed that from their configs.
Attempted to consolidate Actions such that one process handles the details about the Action, regardless of whether it is called by ActionPopup or the Package Action Links - not there yet :(  

### Test Plan
1. Login as a state submitter
2. Navigate to Package Dashboard
3. Click to Waivers tab
4. Locate a Waiver in "Approved" status
5. Click Action kebab, verify "Request Temporary Extension" is an available Action
6. Click "Request Temporary Extension" and verify it goes to the Temporary Extension Form with the correct Parent indicated
7. Return to Package Dashboard
8. Locate Waiver in "Approved" status
9. click Waiver number link to go into the Waiver Details
10. Verify "Request a Temporary Extension" is an available Package Action Link
11. Click "Request a Temporary Extension" and verify it goes to the Temporary Extension Form with the correct Parent indicated

### CODE REVIEW

File: services/app-api/serverless.yml [hunk 1]
```diff
@@ -162,6 +162,27 @@ functions:
           cors: true
           authorizer: aws_iam
 
+  withdrawMedicaidSPA:
```
Comment (bflynn-collabralink) [inline services/app-api/serverless.yml:165]: Nice add!

File: services/ui-src/src/page/base-waiver/BaseWaiverForm.tsx [hunk 2]
```diff
@@ -20,7 +20,6 @@ export const baseWaiverFormInfo: OneMACFormConfig = {
   proposedEffectiveDate: {
     fieldName: "proposedEffectiveDate",
   },
-  actionsByStatus: Workflow.defaultActionsByStatus,
```
[the remark this answers is not in the document]
Comment (kristin-at-theta) [reply services/ui-src/src/page/base-waiver/BaseWaiverForm.tsx:23]: yeah.... though I spent hours trying to make that work - the TypeScript is just not having it... so I might have to delay that until well, at least tomorrow...

[the remark this answers is not in the document]
Comment (kristin-at-theta) [reply services/ui-src/src/page/base-waiver/BaseWaiverForm.tsx:23]: thanks.... I've been trying to move it all the way to the common so the actions are the same for the ellipses, the links on the details page, and the ellipses in the mini dashboards.... I just can't quite get it to connect the way I want...

File: services/common/workflow.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
+
 export const ONEMAC_TYPE = {
   CHIP_SPA: "chipspa",
   CHIP_SPA_RAI: "chipsparai",
   MEDICAID_SPA: "medicaidspa",
   MEDICAID_SPA_RAI: "medicaidsparai",
-  WAIVER: "waiver",
-  WAIVER_BASE: "waivernew",
+  WAIVER: baseWaiver.componentType,
+  WAIVER_BASE: baseWaiver.componentType,
```
Comment (bflynn-collabralink) [inline services/common/workflow.js:9]: Im suspicious that if these are the same do we really need both?

**Review by bflynn-collabralink — APPROVED** [review]
Looks great! I love the consolidation of the popup into a global function instead of needing to define the component (possibly multiple times on the same page) just to get a popup!!

A few things I noticed outside my individual comments:
1. I noticed you moved type definitions from DetailViewDefaults.tsx into DetailLib.ts which is cool but I dont see any commit to remove them from DetailViewDefaults.tsx or remove the file as a whole if everything was moved.
2. I noticed when using the popup from the package view that the table re-renders when the popup is called. This does not currently occur with the existing popup component. Not a big deal but it is a bit disrupting to see the table refresh as the page loads. Might be happening from the details page action too? but it isnt noticeable there since its so quick.

Thanks for all the refactoring and consolidation! I feel like we are much better off with these structural changes!

And oh yeah, the ACs work as expected! :)

Approved!

File: services/ui-src/src/page/section/DetailSection.tsx [hunk 4]
```diff
@@ -115,14 +117,25 @@ export const DetailSection = ({
...
+                                  Workflow.PACKAGE_ACTION.RESPOND_TO_RAI
+                                ? () => {
+                                    onLinkAction({
                                       href: pageConfig.raiLink,
                                       state: {
                                         componentId: detail.componentId,
                                       },
                                     });
                                   }
+                                : () => {
```
[the remark this answers is not in the document]
Comment (kristin-at-theta) [reply services/ui-src/src/page/section/DetailSection.tsx:130]: heh..... yeah - that was where I tried to pop the actions out into something more centralized and well.... it didn't work, so I was in my 4pm brain and that's what you get :) I agree, there is definitely some improvements to be made here

File: services/common/workflow.js (hunk 3 above)
Comment (kristin-at-theta) [reply services/common/workflow.js:9]: Actually, that is the reason I said we should change Medicaid SPA everywhere at once as part of a story.... we use waiver and waivernew as the component type for the base waivers and should definitely clean that up someday :) 

I should probably revert that back to the text, though... I started down a path and it didn't pan out and that piece, while not wrong, is kinda weird to leave like that...

**Pull request merged**